ui/app/AppLayouts/Wallet/controls/FollowingAddressMenu.qml

```diff
@@ -0,0 +1,140 @@
+import "../stores"
+import ".."
+
+import AppLayouts.Wallet.stores as WalletStores
+
+StatusMenu {
+    id: root
+
+    property string name
+    property string address
+    property string ensName
+    property var tags
+    property var activeNetworks
+
+    readonly property int maxHeight: 341
+    height: implicitHeight > maxHeight ? maxHeight : implicitHeight
+
+    QtObject {
+        id: d
+        readonly property string visibleAddress: !!root.ensName ? root.ensName : root.address
+    }
+
+    function openMenu(parent, x, y, model) {
+        root.name = model.name;
+        root.address = model.address;
+        root.ensName = model.ensName;
+        root.tags = model.tags;
+        popup(parent, x, y);
+    }
+
+    onClosed: {
+        root.name = "";
+        root.address = "";
+        root.ensName = ""
+        root.tags = []
+    }
+
+    StatusSuccessAction {
+        id: copyAddressAction
+        objectName: "copyFollowingAddressAction"
+        successText: qsTr("Address copied")
+        text: qsTr("Copy address")
+        icon.name: "copy"
+        timeout: 1500
+        autoDismissMenu: true
+        onTriggered: ClipboardUtils.setText(d.visibleAddress)
+    }
+
+    StatusAction {
+        text: qsTr("Show address QR")
+        objectName: "showQrFollowingAddressAction"
+        assetSettings.name: "qr"
+        onTriggered: {
+            Global.openShowQRPopup({
+                                       showSingleAccount: true,
+                                       showForSavedAddress: false,
+                                       switchingAccounsEnabled: false,
+                                       hasFloatingButtons: false,
+                                       name: root.name,
+                                       address: root.address
+                                   })
+        }
+    }
+
+    StatusAction {
+        text: qsTr("View activity")
+        objectName: "viewActivityFollowingAddressAction"
+        assetSettings.name: "wallet"
+        onTriggered: {
+            Global.changeAppSectionBySectionType(Constants.appSection.wallet,
+                                                 WalletLayout.LeftPanelSelection.AllAddresses,
+                                                 WalletLayout.RightPanelSelection.Activity,
+                                                 {savedAddress: root.address})
+        }
+    }
+
+    StatusMenuSeparator {}
+
+    BlockchainExplorersMenu {
+        id: blockchainExplorersMenu
+        flatNetworks: root.activeNetworks
+        onNetworkClicked: {
+            let link = Utils.getUrlForAddressOnNetwork(shortname, isTestnet, d.visibleAddress ? d.visibleAddress : root.ensName);
+            Global.openLinkWithConfirmation(link, StatusQUtils.StringUtils.extractDomainFromLink(link));
+        }
+    }
+
+    StatusMenuSeparator { }
+
+    StatusAction {
+        text: {
+            var savedAddr = WalletStores.RootStore.getSavedAddress(root.address)
```

Comment:
You should extract the `savedAddr`, `isSaved` etc.. as a `readonly property` of this StatusAction, and evaluate it only once, instead of multiple times for each property here